staging/src/k8s.io/api/autoscaling/v2/types.go

```diff
@@ -361,7 +360,7 @@ type HorizontalPodAutoscalerStatus struct {
 
 	// currentReplicas is current number of replicas of pods managed by this autoscaler,
 	// as last seen by the autoscaler.
-	CurrentReplicas int32 `json:"currentReplicas" protobuf:"varint,3,opt,name=currentReplicas"`
+	CurrentReplicas *int32 `json:"currentReplicas" protobuf:"varint,3,opt,name=currentReplicas"`
```

Comment:
what is the purpose of changing this to a pointer? should it also be made `+optional` and `omitempty`?

Reply:
According to  @josephburnett said "Only one request, to change Status CurrentReplicas to a pointer (to support scale-to-zero). Otherwise it looks good from the Autoscaling SIG perspective.", I changed this to a pointer.

Reply:
What is the difference between omitting this from a status update and setting it to zero? What is the default value if omitted? If the internal and v1 type still has a non-pointer, then there's no ability to distinguish between absent and zero after round-tripping to those versions.

Reply:
if this needs to be able to distinguish between absent and zero, should status.DesiredReplicas also be made a pointer?

Reply:
Ok. I have changed this to int32.

Reply:
@josephburnett how important is it to distinguish between "I am managing 0 replicas" from "I have never recorded status"? The inability to preserve that distinction in all HPA API versions makes that difficult, even if we make this a pointer.

Reply:
No, it's not super important.  And I see your point about conversion .... in v1 where do we put it?  We would have to serialize the additional states to annotations.  So let's not change this now.